src/libs/actions/Session.js

```diff
@@ -208,19 +208,19 @@ function restartSignin() {
  *
  * @param {String} password
  * @param {String} validateCode
+ * @param {String} email
  */
-function setPassword(password, validateCode) {
+function setPassword(password, validateCode, email) {
     Onyx.merge(ONYXKEYS.ACCOUNT, {error: '', loading: true});
 
     API.SetPassword({
-        email: credentials.login,
+        email: email || credentials.login,
         password,
         validateCode,
```

Comment:
wrt my previous comment lets pass accountID as an additional param then the PHP code for setPassword will extract email from the account ID if provided to make this a lot more simpler.